### PR TITLE
Add native support for Elasticsearch REST

### LINKED ISSUE #896: Elasticsearch Rest native mode
A followup of https://github.com/apache/camel-quarkus/pull/895

### PR DESCRIPTION
**Pull request by jamesnetherton**
Fixes #896

### CODE REVIEW

File: tooling/package-maven-plugin/src/main/java/org/apache/camel/quarkus/maven/UpdateDocExtensionsListMojo.java [hunk 1]
```diff
@@ -127,6 +127,11 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                             final DataFormatModel delegate = (DataFormatModel) m.delegate;
                             delegate.setName("bindy");
                         }
+                        // TODO: Fix typo in ES REST metadata. Remove for Camel 3.3.0.
```
Comment (ppalaga) [inline tooling/package-maven-plugin/src/main/java/org/apache/camel/quarkus/maven/UpdateDocExtensionsListMojo.java:130]: Would you please file a followup issue for this?

Comment (jamesnetherton) [reply tooling/package-maven-plugin/src/main/java/org/apache/camel/quarkus/maven/UpdateDocExtensionsListMojo.java:130]: Yep - once this is merged, I'll raise one :+1:

File: integration-tests/elasticsearch-rest/src/test/java/org/apache/camel/quarkus/component/elasticsearch/rest/it/ElasticsearchRestTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.Matchers.is;
+
+@QuarkusTest
+@QuarkusTestResource(ElasticSearchTestResource.class)
+class ElasticsearchRestTest {
+
+    @Test
+    public void loadComponentElasticsearchRest() {
```
Comment (aldettinger) [inline integration-tests/elasticsearch-rest/src/test/java/org/apache/camel/quarkus/component/elasticsearch/rest/it/ElasticsearchRestTest.java:32]: Detail but a better test name could be good if possible.

Comment (jamesnetherton) [reply integration-tests/elasticsearch-rest/src/test/java/org/apache/camel/quarkus/component/elasticsearch/rest/it/ElasticsearchRestTest.java:32]: Whoops nice catch! I forgot to rename that. Unless I have to do another CI cycle on this, I'll do a follow up commit to fix.

**Pull request merged**